src/main/java/org/jenkinsci/plugins/configfiles/folder/FolderConfigFileAction.java

```diff
@@ -21,27 +21,57 @@
 import org.jenkinsci.plugins.configfiles.Messages;
 import org.kohsuke.accmod.Restricted;
 import org.kohsuke.accmod.restrictions.NoExternalUse;
-import org.kohsuke.stapler.*;
+import org.kohsuke.stapler.HttpRedirect;
+import org.kohsuke.stapler.HttpResponse;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.StaplerProxy;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
 
-import javax.servlet.ServletException;
-import java.io.IOException;
-import java.util.*;
+import hudson.Extension;
+import hudson.Util;
+import hudson.model.Action;
+import hudson.model.Item;
+import hudson.model.Job;
+import hudson.security.ACL;
+import hudson.security.Permission;
+import hudson.util.FormValidation;
+
+import com.cloudbees.hudson.plugins.folder.AbstractFolder;
+
+import jenkins.model.TransientActionFactory;
+import net.sf.json.JSONObject;
 
 public class FolderConfigFileAction implements Action, ConfigFilesUIContract, StaplerProxy {
 
-    private Folder folder;
+    private AbstractFolder<?> folder;
 
-    FolderConfigFileAction(Folder folder) {
+    FolderConfigFileAction(AbstractFolder<?> folder) {
         this.folder = folder;
     }
 
-    public Folder getFolder() {
+    public AbstractFolder<?> getFolder() {
         return folder;
     }
 
     @Override
     public String getIconFileName() {
-        return folder.hasPermission(Item.EXTENDED_READ) ? ConfigFilesManagement.ICON_PATH : null;
+        // whilst you may be able to read configuration of a ComputedFolder they are not configurable, so the link should never be shown.
+        boolean hasPerm = folder.hasPermission(Item.EXTENDED_READ);
+        if (hasPerm) {
+            // check that the thing can store configuration (ie it is ultimately configurable)
+            // need to do this as System as a user with EXTENDED_READ but not configure should still see this.
+            // TODO use ACL.as when baseline is 2.14+
+            final SecurityContext impersonate = ACL.impersonate(ACL.SYSTEM);
+            try {
+                hasPerm = folder.hasPermission(Item.CONFIGURE);
+            }
```

Comment:
I am not sure I follow this, will not this be always true since you are impersonating as system?

Reply:
exactly see my above comment.

Reply:
so why even bothering doing it, can you just check `Item.EXTENDED_READ` and then `hasPerm` is true and you show it. Why do you check configure if hasPerm is already `true`

Reply:
because you can have extended_read for say a computed folder.  but no user can ever have configure for that computed folder - so there can never be any managed files for it - so this was the case I want to not show the Action for.  But does not seem to be easily possible so the use case needs to be adapted slightly to only show the action if you can "create new files, or there are some files and you have permission to see their configuration"